cl/lib/utils.py

```diff
@@ -118,3 +118,75 @@ def human_sort(
         sorter = lambda item: [convert(c) for c in re.split("([0-9]+)", item)]
 
     return sorted(unordered_list, key=sorter)
+
+
+def microservice(
```

Comment:
Cool. This works nicely and generalizes things nicely. One thing I'd suggest is that you look at StringIO or BytesIO if you haven't already. If you use those, you can combine the `file` parameter with the `filepath` parameter so that everything calls the microservice the same way.

The other high-level comment I have is that we should find a different home for this function. This file contains tools for manipulating python objects. It's sort of a "I wish Python had this feature" toolbox. Maybe it just goes in `lib/microservice_utils.py`?

Reply:
great.

Reply:
I think everything does work correctly for the micro service.  with regard to files.  I'll update a few tweaks you suggested but, file path, file as bytes or items from the database all get sent over in the same standard manner.